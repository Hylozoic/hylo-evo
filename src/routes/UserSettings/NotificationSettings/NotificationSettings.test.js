import NotificationSettings from './NotificationSettings'
import { shallow } from 'enzyme'
import React from 'react'

describe('NotificationSettings', () => {
  const currentUser = {
    hasDevice: true,
    settings: {
      digestFrequency: 'daily',
      dmNotifications: 'none',
      commentNotifications: 'email'
    }
  }

  it('renders correctly', () => {
    const wrapper = shallow(<NotificationSettings
      currentUser={currentUser}
      updateUserSettings={() => {}} />)
    expect(wrapper).toMatchSnapshot()
  })

  it("hides mobile options if user doesn't have device", () => {
    const wrapper = shallow(<NotificationSettings currentUser={{
      ...currentUser,
      hasDevice: false
    }} />)
    expect(wrapper.find('Select').at(1).prop('options')).toHaveLength(2)
  })
})
