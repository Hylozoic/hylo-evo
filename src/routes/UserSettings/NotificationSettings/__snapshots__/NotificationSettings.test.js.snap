// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSettings renders correctly 1`] = `
<div>
  <div
    data-styleName="title"
  >
    Notifications
  </div>
  <div
    data-styleName="prompt"
  >
    How often would you like to receive email digests for new posts in your communities?
  </div>
  <Select
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "daily",
          "label": "Daily",
        },
        Object {
          "id": "weekly",
          "label": "Weekly",
        },
        Object {
          "id": "never",
          "label": "Never",
        },
      ]
    }
    selected="daily"
  />
  <div
    data-styleName="prompt"
  >
    How would you like to receive notifications about new direct messages?
  </div>
  <Select
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "none",
          "label": "None",
        },
        Object {
          "id": "email",
          "label": "Email",
        },
        Object {
          "id": "push",
          "label": "Mobile App",
        },
        Object {
          "id": "both",
          "label": "Both",
        },
      ]
    }
    selected="none"
  />
  <div
    data-styleName="prompt"
  >
    How would you like to receive notifications about new comments on posts you're following?
  </div>
  <Select
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "none",
          "label": "None",
        },
        Object {
          "id": "email",
          "label": "Email",
        },
        Object {
          "id": "push",
          "label": "Mobile App",
        },
        Object {
          "id": "both",
          "label": "Both",
        },
      ]
    }
    selected="email"
  />
  <div
    data-styleName="help"
  >
    <p
      data-styleName="help-paragraph"
    >
      Download our 
      <a
        href="https://itunes.apple.com/app/appName/id1002185140"
        target="_blank"
      >
        iOS
      </a>
      Â  or 
      <a
        href="https://play.google.com/store/apps/details?id=com.hylo.hyloandroid"
        target="_blank"
      >
        Android
      </a>
       app to receive push notifications.
    </p>
    <p
      data-styleName="help-paragraph"
    >
      See the 
      <Link
        replace={true}
        to="/settings/communities"
      >
        Communities
      </Link>
       section to change notifications for an individual community.
    </p>
  </div>
</div>
`;
